lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +242,93 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        this.exportKeys = Object.keys(this.exports);
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.exportKeys.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+        const methodUnwrapMap = new WeakMap();
+
+        const wrap = (target, name, value) => {
```

Comment:
👆 There should be unit tests for it _(or added if not)_ but the gist of the issue is that native methods _(so not plain or bound ones)_ may be picky about their `this` binding and need wrapping for general use when separated from their object.

```js
process.cwd.call(new Proxy(process, {}))
// TypeError: Illegal invocation
```

Reply:
We should totally file a v8 bug on this though as this does not effect NodeChakra.

Reply:
Amazing, that helps a lot to know! If it seems like it may well be removed by the time this is unflagged, then we really don't need to worry about the perf on this path for now.

Could be worth noting the justification and v8 issue in a comment.

Reply:
I'm 👍 on adding comments. I'll add moar to my implementation too.

Reply:
Once the [V8 bug](https://bugs.chromium.org/p/v8/issues/detail?id=5773) is fixed in addition to dropping the proxy wrapper for native methods we can also conditionally exclude the `get` trap on the `module.exports` proxy for exports that are functions or those we expect to be `[object Object]` _(so likely all builtins)_. Since then the only thing the `get` trap will be used for is correcting `toStringTag` values.